packages/moonstone/ContextualPopupDecorator/ContextualPopupDecorator.js

```diff
@@ -490,10 +490,9 @@ const ContextualPopupDecorator = hoc(defaultConfig, (config, Wrapped) => {
 
 			if (!direction) return;
 
-			const spotlessSpotlightModal = spotlightRestrict === 'self-only' && (
-				Spotlight.getSpottableDescendants(containerId).length === 0
-			);
-			const shouldSpotPopup = current === activator && direction === this.adjustedDirection;
+			const hasSpottables = Spotlight.getSpottableDescendants(containerId).length > 0;
+			const spotlessSpotlightModal = spotlightRestrict === 'self-only' && !hasSpottables;
+			const shouldSpotPopup = current === activator && direction === this.adjustedDirection && hasSpottables;
```

Comment:
Technically line 502 below could be shortened to just `if (!spotlessSpotlightModal) {` because the conditional is redundant with 497.  not changing it, tho.